apps/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,36 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { CommonModule, NgFor } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
+import { City } from '../../model/city.model';
+import { Student } from '../../model/student.model';
+import { Teacher } from '../../model/teacher.model';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgFor, ListItemComponent, CommonModule],
 })
 export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+  @Input() list!: Array<Teacher | City | Student>;
   @Input() customClass = '';
+  @Input('optionTemplate')
+  optionTemplateRef!: TemplateRef<any>;
```

Comment:
yes that's one way to do it. 
You should add the read meta property to target the right element;
```typescript
@Input('optionTemplate', {read: TemplateRef})
  optionTemplateRef!: TemplateRef<any>;
```

Another way is by using contentChild. So, you will declare your template inside your parent selector and this will be easier to read, and to link that the template belong to that element. 

But more importanlty, don't use magic string like `optionTemplate` but use directive:
- you can add typing: [see this article](https://medium.com/@thomas.laforge/directive-type-checking-3d3c5f1137b3)
- no magic string